test-d/readonly-deep.ts

```diff
@@ -0,0 +1,44 @@
+import {expectType, expectError} from 'tsd';
+import {ReadonlyDeep} from '../source/readonly-deep';
+
+const data = {
+	object: {
+		foo: 'bar'
+	},
+	fn: (_: string) => true,
+	string: 'foo',
+	number: 1,
+	boolean: false,
+	symbol: Symbol('test'),
+	null: null,
+	undefined: undefined, // eslint-disable-line object-shorthand
+	map: new Map<string, string>(),
+	set: new Set<string>(),
+	array: ['foo'],
+	tuple: ['foo'] as ['foo'],
+	readonlyMap: new Map<string, string>() as ReadonlyMap<string, string>,
+	readonlySet: new Set<string>() as ReadonlySet<string>,
+	readonlyArray: ['foo'] as ReadonlyArray<string>,
+	readonlyTuple: ['foo'] as const
+};
+
+const readonlyData: ReadonlyDeep<typeof data> = data;
```

Comment:
Any reason of doing this over `const readonlyData = data as ReadonlyDeep<typeof data>;`? Or just a preference?

Reply:
I suppose I was very uninspired when I wrote this. Do you want me to change this? There is no technical difference between the two.

Reply:
Nah, was just curious.